lib/rspec/mocks/error_generator.rb

```diff
@@ -213,6 +213,11 @@ def raise_already_invoked_error(message, calling_customization)
         notify MockExpectationAlreadyInvokedError.new(error_message)
       end
 
+      def raise_expectation_on_nil_error(method_name, source)
+        __raise "An expectation of :#{method_name} was set on nil. Called from #{source}. " \
+          "To allow expectations on nil, set allow_expectations_on_nil to true."
```

Comment:
* There's a fair bit of duplication between this method and `ProxyForNil#warn` (the message text is identical, right?).  WDYT about extracting a `expectation_on_nil_message` helper method that both can delegate to?
* The "Called from source" bit is unnecessary for an exception -- exception carry backtrace info and our exception formatting will include that.  `Kernel.warn` (which is used by `ProxyForNil#warn`) does not include the call site so we were including it manually before.  Since the message was originally written, we've added `RSpec.warning` as a newer standardized way to emit warnings, which takes care of including the source call site in the message...so I think we'll also want `ProxyForNil#warn` updated to use that rather than `Kernel.warn`.